database/redis/tag.go

```diff
@@ -44,6 +44,41 @@ func (connector *DbConnector) GetTagTriggerIDs(tagName string) ([]string, error)
 	return triggerIDs, nil
 }
 
+// CleanUpAbandonedTags deletes tags for which triggers and subscriptions don't exist.
+// Returns count of deleted tags.
+func (connector *DbConnector) CleanUpAbandonedTags() (int, error) {
+	var count int
+	client := *connector.client
+
+	iter := client.SScan(connector.context, tagsKey, 0, "*", 0).Iterator()
+	for iter.Next(connector.context) {
+		tag := iter.Val()
+
+		result, err := client.Exists(connector.context, tagTriggersKey(tag)).Result()
+		if err != nil {
+			return 0, fmt.Errorf("failed to check tag triggers existence, error: %v", err)
```

Comment:
If we fail to check existance of one tag, we stop the whole cleanup? Maybe we should just skip this tag and then somehow return the list of erroneous tags?

Reply:
We return 0, even though we might have deleted some tags, if I understand the code correctly

Reply:
> If we fail to check existance of one tag, we stop the whole cleanup?

Yes. We need to cleanup all abandoned tags. If there is an error, we should fix it and then cleanup all abandoned tags. 

> Maybe we should just skip this tag and then somehow return the list of erroneous tags?
It is nice approach, but I think here it is overhead

Reply:
> We return 0, even though we might have deleted some tags, if I understand the code correctly

Thanks, fixed https://github.com/moira-alert/moira/pull/773/commits/e0cf2e633c291cd7c883ae1e7ccb9bacf91bb685